services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/schema/AnnotationsTranslatorImpl.java

```diff
@@ -367,6 +389,54 @@ void writeAnnotationValue(String key, AnnotationsValue value, JSONObject jsonObj
 			jsonObject.put(key, stringToObject(value.getType(), value.getValue().get(0)));
 		}
 	}
+	
+	/* handle with schema */
+	void writeAnnotationsToJSONObjectWithSchema(Annotations toWrite, JSONObject jsonObject, JsonSchema schema) {
+		Map<String, AnnotationsValue> annotationsMap = toWrite.getAnnotations();
+		// first write with the schema (only writes annotations that appear in the schema as arrays)
+		writeAnnotationValueWithSchema(annotationsMap, jsonObject, schema);
+		// write everything else
+		writeAnnotationsToJSONObject(toWrite, jsonObject);
+	}
+	
+	/* only writes annotations that appear in the schema and are defined as arrays */
+	void writeAnnotationValueWithSchema(Map<String, AnnotationsValue> annotationsMap, 
+			JSONObject jsonObject, JsonSchema schema) {
+		// write the top-level schema first because the SubSchemaIterable does not look at it
+		writeAnnotationsValueWithSchemaHelper(schema, annotationsMap, jsonObject);
+		// look at the subschemas
+		for (JsonSchema subSchema : SubSchemaIterable.depthFirstIterable(schema)) {
+			writeAnnotationsValueWithSchemaHelper(subSchema, annotationsMap, jsonObject);
+		}
+	}
+	
+	void writeAnnotationsValueWithSchemaHelper(JsonSchema schema, Map<String, AnnotationsValue> annotationsMap, 
```

Comment:
1. Crate a map from the schema Map<key,boolean> singleKeys.  No Schema is simply an empty map.
2. Walk annotations, if single then write as single, else array.